### PR TITLE
feat: upgrade power_linux_sap roles

### PR DESCRIPTION
**Pull request by ludwig-mueller**
### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/ansible/templates-ansible/configure-os-for-sap/playbook-configure-os-for-sap.yml.tftpl [hunk 1]
```diff
@@ -15,8 +15,7 @@
     sap_domain: ${sap_domain}
 
   roles:
-  - { role: ibm.power_linux_sap.powervs_prepare_rhel_sap, when: sap_solution is defined and ansible_distribution is match("RedHat*") }
-  - { role: ibm.power_linux_sap.powervs_prepare_sles_sap, when: sap_solution is defined and ansible_distribution is match("SLES*") }
+   - { role: ibm.power_linux_sap.powervs_configure_os_for_sap, when: sap_solution is defined, ignore_errors: true }
```
Comment (surajsbharadwaj) [inline modules/ansible/templates-ansible/configure-os-for-sap/playbook-configure-os-for-sap.yml.tftpl:18]: ignore_errors: this should be removed. Playbook and terraform should fail if any of the tasks fail

File: modules/ansible/templates-ansible/configure-network-services/playbook-configure-network-services.yml.tftpl [hunk 2]
```diff
@@ -13,7 +13,5 @@
     client_config: ${client_config}
 
   roles:
-  - role: ibm.power_linux_sap.powervs_client_enable_services
+  - role: ibm.power_linux_sap.configure_network_management_services
     when: client_config.keys()|length > 0
```
Comment (surajsbharadwaj) [inline modules/ansible/templates-ansible/configure-network-services/playbook-configure-network-services.yml.tftpl:17]: add one more  condition. It should be or condition
server_config.keys()|length > 0

**Review by surajsbharadwaj — CHANGES_REQUESTED** [review]

**Comment by surajsbharadwaj** [comment]
@ludwig-mueller please update the toc-version, last modified and version to 2.2.0. Also do a commit with SKIP UGRADE TEST: landing zone version upgrade"

**Review by surajsbharadwaj — DISMISSED** [review]

**Comment by surajsbharadwaj** [comment]
@ludwig-mueller sync with @stafaniasaju and include powervs_os_registration role and set fls true for checks to be performed.
Also keep in mind how to support BYOL (username and password for custom os images)

File: modules/ansible/templates-ansible/configure-os-for-sap/playbook-configure-os-for-sap.yml.tftpl [hunk 3]
```diff
@@ -15,8 +15,7 @@
     sap_domain: ${sap_domain}
 
   roles:
-  - { role: ibm.power_linux_sap.powervs_prepare_rhel_sap, when: sap_solution is defined and ansible_distribution is match("RedHat*") }
-  - { role: ibm.power_linux_sap.powervs_prepare_sles_sap, when: sap_solution is defined and ansible_distribution is match("SLES*") }
+   - { role: ibm.power_linux_sap.powervs_configure_os_for_sap, when: sap_solution is defined }
 
   tasks:
   - include_role:
```
Comment (surajsbharadwaj) [inline modules/ansible/templates-ansible/configure-os-for-sap/playbook-configure-os-for-sap.yml.tftpl:21]: Lines from 21 to 43 must be removed as v3.0.0 does it already in the role ibm.power_linux_sap.powervs_configure_os_for_sap,

**Review by surajsbharadwaj — CHANGES_REQUESTED** [review]
Lines from 21 to 43 must be removed as v3.0.0 does it already in the role ibm.power_linux_sap.powervs_configure_os_for_sap,

**Comment by ludwig-mueller** [comment]
- PR for powervs_os_registration on instance repo is open, will update this repo to new version once that's merged
- Integration of byol in this repo and in instance needs further discussion

**Review by surajsbharadwaj — APPROVED** [review]

**Pull request merged**